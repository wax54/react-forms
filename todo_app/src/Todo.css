.Todo-title{
    margin: 1rem;
    cursor: pointer;
}