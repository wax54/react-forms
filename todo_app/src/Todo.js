import React from "react";
import './Todo.css';

const Todo = ({ content, onClick, onDelete, id }) => {

    return (
    <div className="Todo">
        <span className="Todo-title" onClick={() => onClick(id)}>{ content }</span>
        <button className="Todo-delete" onClick={ () => onDelete(id) }>X</button>
    </div>
    )
};

export default Todo