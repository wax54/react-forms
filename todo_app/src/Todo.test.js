import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Todo from './Todo';

it("Doesn't Blow up", () => {
    render(<Todo />);
});

it('should match previous snapshot', function () {
    const { asFragment } = render(<Todo content="NEW" id="SOMEID" onDelete={() => undefined} onClick={() => undefined}/>);
    expect(asFragment()).toMatchSnapshot();
});
