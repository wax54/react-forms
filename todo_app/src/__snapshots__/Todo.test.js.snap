// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <span
      class="Todo-title"
    >
      NEW
    </span>
    <button
      class="Todo-delete"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
